exercise/cache/exercise.py

```diff
@@ -28,8 +36,11 @@ def _generate_data(self, exercise, data=None):
                 *self.load_args,
                 last_modified=data['last_modified'] if data else None
             )
-            
-            content = compressor(page.content.encode('utf-8'))
+
+            try:
+                content = compress(page.content.encode('utf-8'), compression=1)
```

Comment:
Hmm, apparently they have a different APIs... That should be handled when defining the functions. Currently, it's not clear if compress raises a TypeError for some other reason.

You can use `functools.partial` or define the functions your self. E.g.:

```python
try:
    from lz4.block import compress as _compress, decompress
    def compress(data):
        return _compress(data, compression=1)
except ImportError:
    logger.warning("Unable to import lz4, using a slower zlib instead")
    from zlib import compress as _compress, decompress
    def compress(data):
        return _compress(data, level=1)
```